Actions - close card and change pos

bgertonson opened the pull request:
Created unique classes for close card and change position actions.

bgertonson committed:
Support for the updateCard:closed action

bgertonson committed:
Different Action Class for Position Changes
 + New UpdateCardPositionAction class
 + Added Pos property to CardName

dillenmeister commented:
I have a question before I merge. Does the card information from trello (CardName) always return Pos and Closed, even if it's not the "update card" or "close card" actions?

bgertonson commented:
No, it does not. Those properties are specific to change position, and close card actions. How would you prefer to have them? The properties could be nullable, which would indicate that they did not come back (instead of the values having the default values). Or, I could expand on the specific action classes instead of simply inheriting from UpdateCardAction, and provide each action class with a specific nested CardName class.

Thoughts?

dillenmeister commented:
I think the second alternative is the best (specific nested classes). The nested classes could inherit from CardName. What do you think?

bgertonson committed:
CloseCard and UpdateCardPosition Action Changes
 + Removed the closed and pos properties from cardname
 + gave each of the actions above a custom cardname class
 + added pos to UpdateCardPositionAction.CardName class
 + added closed to CloseCardAction.CardName class

bgertonson commented:
I pushed a new changeset. I think there are pros and cons to this approach. But the classes don't inherit from UpdateCardAction anymore because they needed to have unique ActionData classes that would use the new nested CardName classes that add the additional parameter. I think the pattern is good, but it does add a lot of additional code to support an additional property.

Any other way I thought to implement the change resulted in the need to cast the CardName to the correct type to get at the additional properties.

dillenmeister commented:
Merged! I moved the new action classes to their own files too.

Good job on this pull request!
